pkgs/build-support/build-typst-package.nix

```
@@ -0,0 +1,35 @@
+{ lib, stdenvNoCC }:
+{
+  nativeBuildInputs ? [ ],
+  buildInputs ? [ ],
+  meta ? { },
+  ...
+}@attrs:
+let
+  typstPkgPrefix = "typst-package";
+in
+stdenvNoCC.mkDerivation (
+  attrs
+  // (with attrs; rec {
+    name = lib.concatStringsSep "-" [
+      typstPkgPrefix
+      pname
+      version
+    ];
```

```suggestion
    name = "typst-package-${pname}-${version}";
```
No reason to make this complicated.